signer/xpi/xpi.go

```diff
@@ -278,3 +278,50 @@ func (sig *Signature) String() string {
 	pem.Encode(&buf, &pem.Block{Type: "PKCS7", Bytes: sig.Data})
 	return string(buf.Bytes())
 }
+
+// verifyPKCS7SignatureRoundTrip checks that
+//
+// 1) the signed XPI includes a PKCS7 signature and signature data
+// 2) the signature serializes and deserializes properly
+// 3) the signature cert chain verifies (domain, expiration date, etc.)
+//
+func verifyPKCS7SignatureRoundTrip(signedFile signer.SignedFile) error {
+	var (
+		sigStr = base64.StdEncoding.EncodeToString(mustReadFileFromZIP(signedFile, "META-INF/mozilla.rsa"))
+		sigData = mustReadFileFromZIP(signedFile, "META-INF/mozilla.sf")
```

Comment:
where is this magical `mustReadFileFromZIP` ?

Reply:
https://github.com/mozilla-services/autograph/pull/89/files#diff-cf69c99e15688b966182afd35812eaf9R174